docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
+
+## Running mypy on Zulip's code
+
+To run mypy on Zulip's code, you'll need to make sure that mypy is installed.
+mypy only runs on python 3. Zulip doesn't use the latest version of mypy available on pypi.
+It uses a specific commit from [mypy's git repository](https://github.com/python/mypy).
+
+*   If you installed Zulip's development environment using vagrant or provision.py,
+    mypy is already installed. To switch to an appropriate python 3 virtual environment, run
+
+        source /srv/zulip-py3-venv/bin/activate
+
+*   If you have not installed Zulip using the above methods, you will have to manually set up
+    a python 3 virtual environment and install mypy in it. Run the following code from Zulip's
+    root directory:
+
+        virtualenv -p python3 ../zulip-py3-venv
+        source ../zulip-py3-venv/bin/activate
+        pip install --upgrade pip
+        pip install -r py3_test_reqs.txt
```

Comment:
To install mypy, we'll first have to install a python 3 virtualenv. I can't put that code in install-mypy since in the future we'll be installing a python 3 virtualenv because we would have added python 3 compatibility to Zulip. So a python 3 venv will have to be installed in provision.py.

So provision.py will still have many lines of code responsible for installing a python 3 venv and install-mypy will just have the line `pip install --no-deps -r py3_test_reqs.txt`.

I don't think adding code to install-mypy which checks if `zulip-py3-venv` has been installed and installs it if it was not present would be useful. Those who don't use provision.py will install a venv for python 2 themselves. They might as well install a python 3 venv themselves.

What are your thoughts on this?